source/common/vb_gui.c

```diff
@@ -343,14 +347,38 @@ int emulation_reset(void) {
     return D_EXIT;
 }
 
-int emulation_sstate(void) {
+char * get_savestate_path(int state) {
+    char * sspath;
+    sspath = (char *) malloc(131 * sizeof(char));
```

Comment:
Where does the 131 come from? Is it ensured that the following `strcat` do not write beyond the end?
What speaks against the use of a single `snprintf` that assembles the complete path